src/Form/PrintArticleSwitchTypeForm.php

```diff
@@ -0,0 +1,141 @@
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'PrintArticle_switchType';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, $print_article = NULL) {
+
+    $print_article = $this->entityTypeManager->getStorage('print_article')
+      ->load($print_article);
+
+    if ($print_article) {
+
+      /** @var \Drupal\thunder_print\Entity\PrintArticleTypeInterface $print_article_type */
+      $print_article_type = $this->entityTypeManager->getStorage('print_article_type')
+        ->load($print_article->bundle());
+
+      $options = [];
+      /** @var \Drupal\thunder_print\Entity\PrintArticleTypeInterface $entity */
+      foreach ($print_article_type->getSwitchableBundles() as $entity) {
+        $options[$entity->id()] = $entity->label();
+      }
+
+      if ($options) {
+        $form['new_print_type'] = [
+          '#type' => 'select',
+          '#title' => $this->t('New snippet template'),
+          '#options' => $options,
+          '#required' => TRUE,
+        ];
+
+        $form['submit'] = [
+          '#type' => 'submit',
+          '#value' => $this->t('Switch template'),
+        ];
+        $form['print_article'] = [
+          '#type' => 'hidden',
+          '#value' => $print_article->id(),
+        ];
+      }
+      else {
+        $form['error']['#markup'] = $this->t('No matching snippet templates.');
+      }
+
+    }
+    else {
+      $form['error']['#markup'] = $this->t('No valid print article.');
+    }
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+
+    $new_print_type = $form_state->getValue('new_print_type');
+
+    $print_article = $this->entityTypeManager->getStorage('print_article')
+      ->load($form_state->getValue('print_article'));
+
+    /** @var \Drupal\thunder_print\Entity\PrintArticleTypeInterface $print_type */
+    $print_type = $this->entityTypeManager->getStorage('print_article_type')
+      ->load($print_article->bundle());
+
+    $possibleBundles = array_keys($print_type->getSwitchableBundles());
+
+    if (!in_array($new_print_type, $possibleBundles) || !$print_article) {
+      $form_state->setErrorByName('new_print_type', $this->t('Not a valid print article type.'));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+
+    $new_print_type = $form_state->getValue('new_print_type');
+
+    $print_article = $this->entityTypeManager->getStorage('print_article')
+      ->load($form_state->getValue('print_article'));
+
+    if ($print_article && $new_print_type) {
+      $print_article->type = $new_print_type;
```

Comment:
Maybe we should add a new revision that indicates the template switch, although the type is not revisionable by now.